components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java

```diff
@@ -130,9 +130,31 @@ public OAuth2ClientValidationResponseDTO validateClientInfo(String clientId, Str
         }
 
         try {
+            if (StringUtils.isBlank(clientId)) {
+                throw new InvalidOAuthClientException("Invalid client_id. No OAuth application has been registered " +
+                        "with the given client_id");
+            }
             OAuthAppDAO oAuthAppDAO = new OAuthAppDAO();
```

Comment:
We should avoid calling the DAO directly since it's not a cached back DAO. For the moment we have to rely on using the OAuth2Util.getAppInformationByClientId(clientId) method which makes use of the cache.

Reply:
The same class has getOauthApplicationState(String consumerKey) method which does this. Let's reuse that here.